src/statisticBlock.js

```diff
@@ -97,69 +98,103 @@ class StatisticBlock extends Component {
-      let clientWidth = this.state.clientWidth;
-      let clientHeight = this.state.clientHeight;
-      let childHeight = 0;
-
-      if(element.clientHeight == element.scrollHeight
-      && this.state.size == this.props.options.size
-      && !this.state.overflow) return;
-
-      if(this.refs['child-0']) {
-        childHeight = ReactDOM.findDOMNode(this.refs['child-0']).clientHeight;
-      }
+      if(this.props.kpis.qDimensionInfo.length > 0) {
+        let containerElm = this.refs.parent;
+
+        let elementClientWidth = containerElm.getBoundingClientRect().width;
+        let elementClientHeight = containerElm.getBoundingClientRect().height;
+        let childrenHeight = 0;
+        let childrenElm =containerElm.children;
+        let childrenCombinedWidth = 0;
+        let dividedBy =this.props.options.divideBy;
+        let dividedByNumber = getDivideByNumber(dividedBy);
 
-      if(element
-        && ((element.clientHeight < scrollHeight
-          || childHeight && element.clientHeight < childHeight)
-        || ((clientWidth != element.clientWidth
-          || clientHeight != element.clientHeight)
-           && size != this.props.options.size)
-        ))
-      {
-        if(element.clientHeight < scrollHeight
-          || element.clientHeight < childHeight
-          || element.clientWidth < scrollWidth) {
-          if(this.state.size == SIZE_OPTIONS[0].value
-          && this.state.overflow === "auto")
-            return;
 
+        if (this.props.options.dimShowAs == "segment" && this.props.options.dimensionsOrientation == "vertical"){
+
+          childrenCombinedWidth = childrenElm[0].getBoundingClientRect().width;
+          for (let i = 0 ; i < childrenElm.length ; i++){
+            childrenHeight = childrenHeight + childrenElm[i].getBoundingClientRect().height;
+          }
+        }else{
+          if (dividedBy == "auto" || dividedBy == ""){
+            for (let e of childrenElm) {
+              childrenCombinedWidth = childrenCombinedWidth + e.getBoundingClientRect().width;
+            }
+          }else{
+            let rows = this.props.kpis.qMeasureInfo.length;
+            let ratio = Math.floor(rows / dividedByNumber) + 1;
+            if (rows % dividedByNumber == 0){
+              for (let i = 0 ; i < rows/dividedByNumber ; i++ + ratio){
+                childrenHeight = childrenHeight + childrenElm[i].getBoundingClientRect().height;
+              }
+            }
+            else{
+              for (let i = 0 ; i <= Math.floor(rows/dividedByNumber) ; i++ + ratio){
+                childrenHeight = childrenHeight + childrenElm[i].getBoundingClientRect().height;
+              }
+            }
+            for ( let i = 0 ; i < dividedByNumber ; i++){
+              childrenCombinedWidth = childrenCombinedWidth + childrenElm[i].getBoundingClientRect().width;
+            }
+          }
+        }
+        if (elementClientWidth < childrenCombinedWidth || elementClientHeight < childrenHeight){
```

Comment:
This whole section above should be extracted to its own function for getting height/width sizes.